### PR TITLE
💚 Run CAPI E2E specs in E2E

### LINKED ISSUE #699: 3 control plane E2E test is flaky
/kind bug

**What steps did you take and what happened:**
[A clear and concise description of what the bug is.]

See https://testgrid.k8s.io/sig-cluster-lifecycle-cluster-api-provider-azure#pr-e2e for recent history. `Workoad cluster creation Create multiple controlplane cluster with machine deployments Should create a 3 node cluster` has failed multiple times recently. The test times out at 600s with 2 control planes ready out of 3.

**Environment:**

- cluster-api-provider-azure version: 
- Kubernetes version: (use `kubectl version`): 
- OS (e.g. from `/etc/os-release`):

### PR DESCRIPTION
**Pull request by CecileRobertMichon**
**What this PR does / why we need it**: Adds some of the [CAPI E2E specs](https://github.com/kubernetes-sigs/cluster-api/tree/master/test/e2e) to the CAPZ test suite. These won't be run as part of presubmit (PR tests) for now, only optionally when `/test pull-cluster-api-provider-azure-capi-e2e` is commented on the PR.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #699

**Special notes for your reviewer**: For now the tests that were failing are commented out and will be enabled as part of a later PR.

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [x] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
Run CAPI E2E specs in E2E
Use the prow template with tags for E2E tests
```

### CODE REVIEW

File: test/e2e/capi_test.go [hunk 1]
```diff
@@ -0,0 +1,86 @@
+/*
```
Comment (nader-ziada) [inline test/e2e/capi_test.go:1]: missing the tag `// +build e2e` in the beginning of the file

**Comment by CecileRobertMichon** [comment]
/assign @chewong @nader-ziada @fabriziopandini 

I disabled the tests that were failing and will investigate those separately. For now, I'd like to merge this to add the skeleton and fix #699 + be able to add the periodic job to prow.

File: test/e2e/config/azure-dev.yaml [hunk 2]
```diff
@@ -42,13 +42,18 @@ providers:
     type: InfrastructureProvider
     versions:
       - name: v0.4.0
-        value: ../../../config
+        value: "${PWD}/config"
     files:
-      - sourcePath: ../../../templates/cluster-template.yaml
+      - sourcePath: "${PWD}/templates/test/cluster-template-prow.yaml"
```
Comment (CecileRobertMichon) [inline test/e2e/config/azure-dev.yaml:47]: @chewong mostly want your 👀 on this ^

**Comment by fabriziopandini** [comment]
LGTM from my side. Looking forward to see this moving

File: test/e2e/config/azure-dev.yaml (hunk 2 above)
Comment (chewong) [reply test/e2e/config/azure-dev.yaml:47]: LGTM

**Comment by nader-ziada** [comment]
e2e passed 🎉🎉

**Comment by CecileRobertMichon** [comment]
@nader-ziada can I get a /lgtm?

**Comment by nader-ziada** [comment]
@CecileRobertMichon great job on this PR

**Comment by CecileRobertMichon** [comment]
/approve

Thanks @nader-ziada !

**Pull request merged**